lib/pokerscore/compare_hand_types.rb

```diff
@@ -1,12 +1,16 @@
-require 'const/hand_constants'
 require 'lib/pokerscore/texas_hold_em/identify'
 
 class CompareHandTypes
+
+  def initialize(hand_values)
+    @hand_values = hand_values
+  end
+
   def call(hand1, hand2)
     identify = TexasHoldEm::Identify.new
-    hand1_type = $hand_types[identify.call(hand1)]
-    hand2_type = $hand_types[identify.call(hand2)]
-    case hand1_type <=> hand2_type
+    value1 = @hand_values.get_value(identify.call(hand1))
+    value2 = @hand_values.get_value(identify.call(hand2))
+    case value1 <=> value2
```

Comment:
I might use a more descriptive name than `identify`, as we currently need to refer back to line 10 to understand what this variable represents. Perhaps `identify_hand_type`?

The naming of this becomes even more important if we decide to DI (dependency inject) this later on.

I would also suggest we try and get away from `value1`, and `value2`, as these names have little meaning on their own. Perhaps `hand_1_type_value` and `hand_2_type_value`. Admittedly those names aren't super elegant, but they communicate what the variables represent.